Source/Particles/PulsarParameters.cpp

```diff
@@ -0,0 +1,91 @@
+#include <PulsarParameters.H>
+#include <AMReX_ParmParse.H>
+#include <AMReX_Print.H>
+#include <AMReX_RealVect.H>
+#include <AMReX_REAL.H>
+#include <AMReX_GpuQualifiers.H>
+
+namespace PulsarParm
+{
+   std::string pulsar_type;
+
+   AMREX_GPU_DEVICE_MANAGED amrex::Real omega_star;
+   AMREX_GPU_DEVICE_MANAGED amrex::Real B_star;
+   AMREX_GPU_DEVICE_MANAGED amrex::Real R_star;
+   AMREX_GPU_DEVICE_MANAGED amrex::Real dR_star;
+   AMREX_GPU_DEVICE_MANAGED int EB_external = 0;
+   AMREX_GPU_DEVICE_MANAGED int E_external_monopole = 0;
+   AMREX_GPU_DEVICE_MANAGED amrex::Vector<amrex::Real> center_star(3,0.0);
+   AMREX_GPU_DEVICE_MANAGED int verbose = 0;
+
+   void ReadParameters() {
+      amrex::ParmParse pp("pulsar");
+      pp.query("pulsarType",pulsar_type);
+      pp.query("omega_star",omega_star);
+      pp.queryarr("center_star",center_star);
+      pp.query("R_star",R_star);
+      pp.query("B_star",B_star);
+      pp.query("dR",dR_star);
+      pp.query("verbose",verbose);
+      pp.query("EB_external",EB_external);
+      pp.query("E_external_monopole",E_external_monopole);
+      amrex::Print() << " Pulsar center: " << center_star[0] << " " << center_star[1] << " " << center_star[2] << "\n";
+      amrex::Print() << " Pulsar omega: " << omega_star << "\n";
+      amrex::Print() << " Pulsar B_star : " << B_star << "\n";
+   }
+
+   void PulsarEBField(amrex::Real xp, amrex::Real yp, amrex::Real zp,
```

Comment:
We'll need to add `AMREX_GPU_HOST_DEVICE AMREX_INLINE` right above the line where `PulsarEBField` is declared so we'll be able to call it on the GPU inside the ParallelFor lambda.

```
AMREX_GPU_HOST_DEVICE AMREX_INLINE
void PulsarEBField(...
```

Reply:
done